Fixed aria props getting overriden

wakooka opened the pull request:
## Description

Everytime we were trying to pass `aria-label` to `SlugLink` in `place.href` the prop was getting overriden by the `pickA11yProps`. See here for more context: https://github.com/blablacar/kairos/pull/4487#discussion_r529610375
~We also reproduced it with `ItemChoice` `href` prop.~

As a long term fix we might want to replace `a11yProps` with `React.AriaAttributes` instead

@PerrineBoissieres We need your input on this.

## What has been done

Moved a11y props first so it doesn't override the aria-label passed 

## Things to consider

This is a quick fix, I don't think other things will breaki

## How it was tested

~In storybook by passing a component with `aria-label` in `ItemChoice` `href` prop.~
Unit test + Storybook

pith commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@blablacar/ui-library",
-  "version": "41.9.0",
+  "version": "0.0.0-ed463c5a",
```
I think it's a miss 😉

pith commented on src/itinerary/Itinerary.story.mdx:
```diff
@@ -6,6 +6,11 @@ import { Itinerary } from './index'
 
 <Meta title="Widgets/Itinerary" />
 
+export const MyComponent = props => {
```
You might want to rollback this change?

pith commented:
I would be nice to add a unit test to avoid regression on this cases.

wakooka commented:
EDIT: We might not need the change on `ItemChoice` this was because we were trying to pass `aria-label` in `href` props while we could just pass it as a top level prop, e.g.
```
<ItemChoice aria-label="My aria label" href={<SlugLink  ... />}
```
I'll re-request your reviews once I double checked that it's working in SPA

simonrelet commented on src/_internals/itineraryLocation/ItineraryLocation.tsx:
```diff
@@ -107,11 +107,13 @@ export const ItineraryLocation = ({
       ...link.props,
       className: cc([`${baseClassName}-wrapper`, link.props.className]),
     }
+    console.log(hrefProps)
```
Probably a left over 😊

- [ ] Remove `console.log()`

wakooka replied:
haha just pushed, thanks! :)

wakooka commented:
I updated the PR and removed the change on `ItemChoice` I confirm this was not needed I also updated the tests in `ItineraryLocation` making sure that they will break if change the order of 
```
<Component {...hrefProps} aria-label={place.actionAriaLabel}>
```

pith commented on CHANGELOG.md:
```diff
@@ -1,6 +1,6 @@
 # Unreleased
 
-- [...]
+- **[FIX]** Stopped `aria-*` props from getting overriden in `Item`
```
```suggestion
- **[FIX]** Stopped `aria-*` props from getting overriden in `ItineraryLocation`
```

wakooka replied:
Thanks!
